### PR TITLE
remove buildout

### PR DESCRIPTION
**Pull request by m-kharbat**
## Summary of the changes / Why this is an improvement
 
this pr removes buildout and installs node locally using nodeenv 

## Checklist

 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -61,11 +53,9 @@ def read(path):
...
       namespace_packages=[],
       include_package_data=True,
-      extras_require=dict(
-          test=test_requires,
-      ),
+      extras_require={},
       zip_safe=False,
-      install_requires=requires,
-      tests_require=test_requires,
+      install_requires=[],
```
Comment (chaudum) [inline setup.py]: You can omit the keyword arguments if they are "empty":
* extras_require
* install_requires
* tests_equire

File: setup.py [hunk 2]
```diff
@@ -23,9 +23,9 @@
 from setuptools import setup, find_packages
 import json
 
-bowerJson = json.loads(open("bower.json").read())
-if bowerJson:
-    version = bowerJson.get('version')
+packageJson = json.loads(open("package.json").read())
```
Comment (chaudum) [inline setup.py]: Please close filepointer correctly, e.g. using contextmanager:

```py
def read(path: str) -> str:
    p = Path(os.path.dirname(__file__)) / path
    with open(p.resolve(), "r", encoding="utf-8") as fp:
        return fp.read()

package = json.loads(read("package.json"))
```

File: requirements-docs.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# packages for local dev
+sphinx-autobuild
+
+# the next section should exact copy of the hosted RTD environment (please
+# update if you notice that this is not the case)
+
+alabaster>=0.7,<0.8,!=0.7.5
+setuptools<41
+sphinx==1.7.4
+sphinx-argparse==0.2.5
```
Comment (chaudum) [inline requirements-docs.txt]: `sphinx-argparse` and `.` are only required if the docs use the argparse plugin to document the argument parser.

File: requirements-docs.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+crate-docs-theme>=0.5.74
```
Comment (chaudum) [inline requirements-docs.txt]: Think the version shall not be nailed.

File: setup.py (hunk 2 above)
Comment (chaudum) [reply setup.py]: Also, remove `read()` on line 37 then.

File: setup.py [hunk 5]
```diff
@@ -61,11 +56,8 @@ def read(path):
...
       include_package_data=True,
-      extras_require=dict(
-          test=test_requires,
-      ),
       zip_safe=False,
-      install_requires=requires,
-      tests_require=test_requires,
       test_suite="",
       )
+
```
Comment (chaudum) [inline setup.py]: Please remove empty lines at end of file.

File: setup.py [hunk 6]
```diff
@@ -23,9 +23,10 @@
 from setuptools import setup, find_packages
 import json
 
-bowerJson = json.loads(open("bower.json").read())
-if bowerJson:
-    version = bowerJson.get('version')
+
+packageJson = json.loads(read("package.json"))
```
Comment (chaudum) [inline setup.py]: We use snake_case in Python :)

```suggestion
package_json = json.loads(read("package.json"))
```

File: DEVELOP.rst [hunk 7]
```diff
@@ -10,21 +10,24 @@ You will need Python 3.5 installed.
 Setup
 =====
 
-Bootstrap the project::
+Installing node locally::
 
-    $ python bootstrap.py
-    $ bin/buildout -N
+    $ /usr/bin/python3 -m venv env
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ python3 -m venv env
```

the bin location will vary. this doesn't work on my machine

File: DEVELOP.rst [hunk 8]
```diff
@@ -10,21 +10,24 @@ You will need Python 3.5 installed.
...
+    $ source env/bin/activate
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
```

there's no need to activate. you can just call the binaries directly from the env

File: DEVELOP.rst [hunk 9]
```diff
@@ -10,21 +10,24 @@ You will need Python 3.5 installed.
...
+    $ python -m pip install -U pip
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ env/bin/pip install -U pip
```

File: DEVELOP.rst [hunk 10]
```diff
@@ -10,21 +10,24 @@ You will need Python 3.5 installed.
...
+    $ python -m pip install nodeenv
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ env/bin/pip install nodeenv
```

File: DEVELOP.rst [hunk 11]
```diff
@@ -10,21 +10,24 @@ You will need Python 3.5 installed.
...
+    $ nodeenv  --node=10.16.0 -p
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ env/bin/nodeenv --node=10.16.0 -p
```

File: DEVELOP.rst [hunk 12]
```diff
@@ -123,15 +126,20 @@ To create a new release, you must:
 Writing Documentation
 =====================
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
```

File: DEVELOP.rst [hunk 13]
```diff
@@ -123,15 +126,20 @@ To create a new release, you must:
...
+The docs live under the ``docs/`` directory.
 
-Build the docs by running::
+The docs are written with ReStructuredText_ and processed with Sphinx_.
 
-    $ bin/sphinx
+First, install the additional dependencies by running::
 
-The output can then be found in the ``out/html`` directory.
+    $ pip install -Ur requirements-docs.txt
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ env/bin/pip install -Ur requirements-docs.txt
```

File: DEVELOP.rst [hunk 14]
```diff
@@ -123,15 +126,20 @@ To create a new release, you must:
...
+
+Then build the documentation by running::
+
+    $ env/bin/sphinx-build -b html docs out
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
    $ env/bin/sphinx-autobuild docs out
```

File: DEVELOP.rst [hunk 15]
```diff
@@ -123,15 +126,20 @@ To create a new release, you must:
...
+
+The output can then be found in the ``/out/html/`` directory.
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
```

File: DEVELOP.rst [hunk 16]
```diff
@@ -123,15 +126,20 @@ To create a new release, you must:
...
-    $ bin/sphinx
+First, install the additional dependencies by running::
 
-The output can then be found in the ``out/html`` directory.
+    $ pip install -Ur requirements-docs.txt
+
+Then build the documentation by running::
+
+    $ env/bin/sphinx-build -b html docs out
+
```
Comment (mechanomi) [inline DEVELOP.rst]:
```suggestion
```

**Pull request merged**